### DISCUSSION TITLE
Prisma Seed: Connecting array of items

### DISCUSSION THREAD

**Discussion by BorisBertelsen** [prisma/prisma#22554] [Q&A]
### Question

I'm trying to create:
- A user
- With a crew type associated (driver, photographer etc.)
- A rating of how good they are at each type of job

My question is:
How do I create multiple crewTypes with role and rating, and then connect them to my user model?
Eg. My user can be a Photographer with a rating of 5 but also a driver with the rating of 1.

I have tried something like this but I can't MAP over the crewTypes:
```js
const crewTypeToUser = await prisma.crewTypeToUser.create({
    data: {
      user: { connect: { id: user.id } },
      crewType: {
        connect: crewTypes.map((crewType) => ({ id: crewType.id })),
      },
    },
  });
```

My seed file looks like this so far:
```js
const user = await prisma.user.upsert({
    where: { email: "email@duck.com" },
    update: {},
    create: {
      name: "John Doe",
    },
  });

 const crewTypes = await prisma.crewTypes.createMany({
    data: [
      {
        rating: 5,
        role: "Photographer",
      },
      {
        rating: 1,
        role: "Driver",
      },
    ],
  });

```
My schema looks like this:
```prisma
model User {
  id          Int              @id @default(autoincrement())
  name        String
  crewTypes   CrewTypeToUser[]
}

enum CrewRoles {
  FOTOGRAF
  CHAUFFOER
  LYDTEKNIKER
}

model CrewTypes {
  id     Int              @id @default(autoincrement())
  users  CrewTypeToUser[]
  role   CrewRoles        @default(FOTOGRAF)
  rating Int
}

model CrewTypeToUser {
  userId     Int
  user       User       @relation(fields: [userId], references: [id])
  crewTypeId Int
  crewType   CrewTypes? @relation(fields: [crewTypeId], references: [id])

  @@id([userId, crewTypeId])
}
```

**Comment by ludralph** [comment]
Hi @BorisBertelsen 👋 

Thank you for raising this question.

From your seed file, you are correctly creating the `user` and `crewTypes` with `roles` and `rating`. To create the connection between the `crewType` and `user`, you can do:
```js
for (const crewType of crewTypes) {
  await prisma.crewTypeToUser.create({
    data: {
      user: { connect: { id: user.id } },
      crewType: { connect: { id: crewType.id } },
    },
  });
}
```

We iterate through the `crewTypes` array using a `for...of` loop. For each `crewType`, create a `crewTypeToUser` record using `prisma.crewTypeToUser.create`. Finally, use the `connect` property to establish the relationship between the `user` and the `crewType`, referencing their respective IDs.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
